docs/develop/create-sso-office-add-ins-nodejs.md

```diff
@@ -124,453 +126,477 @@ This article walks you through the process of enabling single sign-on (SSO) in a
-            //          the Office document.
-
-        })
-        .fail(function (errorResult) {
-            showMessage("Error from Microsoft Graph: " + JSON.stringify(errorResult));
-        });
+    try {
+      await ajaxCallToRESTApi(clientRequest);
+    } catch (error) {
+      if (error.statusText === "Internal Server Error") {
+        const isTokenExpired = handleWebServerErrors(error);
+        if (isTokenExpired && clientRequest.authSSO) {
+          try {
+            clientRequest.accessToken = await getAccessTokenFromSSO(clientRequest.authOptions);
+            await ajaxCallToRESTApi(clientRequest);
+          } catch {
+            // If still an error go to fallback.
+            switchToFallbackAuth(clientRequest);
+            return;
+          }
+        } else {
+          // For unhandled errors using SSO, switch to fallback.
+          if (clientRequest.authSSO) {
+            switchToFallbackAuth(clientRequest);
+          } else {
+            console.log(JSON.stringify(error)); // Log any errors.
+            showMessage(error.responseText);
+          }
+        }
+      } else {
+        console.log(JSON.stringify(error)); // Log any errors.
+        showMessage(error.responseText);
+      }
     }
     ```
 
-1. Replace `TODO 10` with the following. About this code, note:
+1. In the `ajaxCallToRESTApi` function, replace `TODO 9` with the following code. About this code, note:
 
-    * This object is the parameter to the `$.ajax` method.
-    * The `/getuserdata` is an Express route on the add-in's server that you create in a later step. It will call a Microsoft Graph endpoint and include the access token in its call.
+    * The function explicitly rethrows any errors for the caller to handle.
 
     ```javascript
-    {
+    try {
+      await $.ajax({
         type: "GET",
-        url: "/getuserdata",
-        headers: {"access_token": accessToken },
-        cache: false
+        url: clientRequest.url,
+        headers: { Authorization: "Bearer " + clientRequest.accessToken },
+        cache: false,
+        success: function (data) {
+          result = data;
+          // Send result to the callback handler.
+          clientRequest.callbackRESTApiHandler(result);
+        },
+      });
+    } catch (error) {
+      // This function explicitly requires the caller to handle any errors
+      throw error;
     }
     ```
 
-1. Replace `TODO11` with the following. About this code, note:
+1. In the `handleWebServerErrors` function, replace `TODO 10` with the following code. About this code, note:
 
-    * The `writeFileNamesToOfficeDocument` will insert the data from Graph into the Office document. It is defined in the `public\javascripts\document.js` file.
-    * If `writeFileNamesToOfficeDocument` returns an error, it will begin with "Unable to add filenames to document."
+    * The error is returned by the middle tier server which will indicate the type of error to make it easier to handle here.
```

Comment:
Sometimes "middle-tier server" has a dash when and other times it doesn't. Could it be consistent? Always one way or the other way 
```suggestion
    * The error is returned by the middle-tier server which will indicate the type of error to make it easier to handle here.
```